src/api_server/src/parsed_request.rs

```diff
@@ -353,95 +353,100 @@ pub(crate) mod tests {
     #[test]
     fn test_error_into_response() {
         // Generic error.
-        let mut buf: [u8; 150] = [0; 150];
+        let mut buf = Cursor::new(vec![0]);
         let response: Response =
             Error::Generic(StatusCode::BadRequest, "message".to_string()).into();
-        assert!(response.write_all(&mut buf.as_mut()).is_ok());
+        assert!(response.write_all(&mut buf).is_ok());
+        let body = ApiServer::json_fault_message("message");
         let expected_response = format!(
             "HTTP/1.1 400 \r\n\
              Server: Firecracker API\r\n\
              Connection: keep-alive\r\n\
              Content-Type: application/json\r\n\
-             Content-Length: 32\r\n\r\n\
+             Content-Length: {}\r\n\r\n\
              {}",
-            ApiServer::basic_json_body("fault_message", "message")
+            body.len(),
+            body
         );
-        assert_eq!(&buf[..], expected_response.as_bytes());
+        assert_eq!(&buf.get_ref()[..], expected_response.as_bytes());
```

Comment:
```suggestion
        assert_eq!(buf.into_inner(), expected_response.as_bytes());
```
it helps in terms of readability, same comment for the entire test.

Reply:
Thanks! I didn't know that. Addressed in 850d689.